### PR TITLE
[TIMOB-25650] Update to babel 7 ecosystem

### PR DESCRIPTION
**Pull request by sgtcoolguy**
https://jira.appcelerator.org/browse/TIMOB-25650

- Update to babel 7 ecosystem. 
- Add quick unit test that polyfilling works.
- Replace wrench usage with fs-extra.
- Bump version to 0.4.14

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
@ewanharris Note that I have not yet tried dropping this into the SDK to see if it breaks it. I simply ran the existing tests and adde one new one for polyfilling. (the tests are not particularly comprehensive here, I'd be happy if anyone had some more tests to write - maybe just a number of files to pass through jsanalyze with different options to verify the output?).

File: lib/jsanalyze.js [hunk 1]
```diff
@@ -196,14 +196,16 @@ exports.analyzeJs = function analyzeJs(contents, opts) {
 			}
 
 			// copy over polyfill and its dependencies
-			// WARNING: REMEMBER TO UPDATE THIS IF 'babel-polyfill' DEPENDENCIES CHANGE!
-			[ 'babel-polyfill', 'core-js', 'regenerator-runtime' ].forEach((moduleName) => {
-				const moduleSrcDir = path.dirname(require.resolve(path.join(moduleName, 'package.json'))),
-					moduleDstDir = path.join(modulesDir, moduleName);
+			// WARNING: REMEMBER TO UPDATE THIS IF '@babel/polyfill' DEPENDENCIES CHANGE!
+			// Because core-js is under @babel/polyfill, it will get copied. require.resolve didn't work here for it anyways
+			[ '@babel/polyfill', 'regenerator-runtime' ].forEach((moduleName) => {
```
Comment (garymathews) [inline lib/jsanalyze.js]: Should we move this outside of the `if (opts.transpile) {` and add some logic to include `core-js` to always polyfill?

**Comment by sgtcoolguy** [comment]
Sort of related: appcelerator/titanium_mobile#10112

File: lib/jsanalyze.js (hunk 1 above)
Comment (ewanharris) [inline lib/jsanalyze.js]: core-js residing under `@babel/polyfill` seems to be causing issues for me in kitchensink as the require path is `core-js/modules/es7.symbol.async-iterator` which isn't the location as it's under `@babel/polyfill`, nuking the package-lock makes puts core-js back in the top level (but causes a whole host of changes).

I think something like the below for resolving the modules might work, but would require us to bump our min Node to 8.9 for the paths option in require.resolve (so we know we're getting the exact dependency `@babel/polyfill` wants), this makes kitchensink happy.

```js
const polyfillPkgJSONKLocation = require.resolve(path.join('@babel/polyfill', 'package.json'));
const polyfillPath = path.dirname(polyfillPkgJSONKLocation);
const polyfillPkgJSON = fs.readJSONSync(polyfillPkgJSONKLocation);
const polyfillDest =  path.join(modulesDir, '@babel/polyfill');
if (!fs.existsSync(polyfillDest)) {
	fs.copySync(polyfillPath, polyfillDest);
}
for (const dependency of Object.keys(polyfillPkgJSON.dependencies)) {
	const dependencyPath = path.dirname(require.resolve(dependency, { paths: [ polyfillPath ] }));
	const moduleDstDir = path.join(modulesDir, dependency);
	if (!fs.existsSync(moduleDstDir)) {
		fs.copySync(dependencyPath, moduleDstDir);
	}
}
```

**Comment by sgtcoolguy** [comment]
@ewanharris I did not pin the babel versions but I did take your polyfill dependency copying suggestion and implemented it with recursion. I had to tweak what you had somewhat for it to work, though - and had to bump the min node to 8.9.0.

**Comment by ewanharris** [comment]
@sgtcoolguy, I'm just catching up on things but will get round to this as soon as I can. What timeframe do you think we're looking to land this in, babel shipped what they hope was the last beta last week, and are hoping to go RC->GA fairly quick (https://github.com/babel/babel/releases/tag/v7.0.0-beta.56). Do we want to land it pre-GA or hold off til GA (maybe landing around the 8.0.0 timeframe?)

**Comment by sgtcoolguy** [comment]
@ewanharris I've updated this PR. It now uses the babel 7.0.0 GA version. It also bumps the node minimum to 8.9.0 and therefore bumps the version to 1.0.0 of this package.

**Pull request merged**